crates/sui-types/src/temporary_store.rs

```diff
@@ -830,12 +830,7 @@ impl<S: ObjectStore> TemporaryStore<S> {
             ))
         } else {
             // else, this is a dynamic field, not an input object
-            if let Ok(Some(old_obj)) = self.store.get_object(id) {
-                if old_obj.version() != expected_version {
-                    return Err(ExecutionError::invariant_violation(
-                        "Expected to find old object with version {expected_version}",
-                    ));
-                }
+            if let Ok(Some(old_obj)) = self.store.get_object_by_key(id, expected_version) {
```

Comment:
This must exist - should we expect / panic if it does not exist?

Reply:
Agreed--I would still return an invariant violation if this is `None`. We will violate conservation otherwise

Reply:
I am removing those change considering they are in gas legacy code